CharlieBackend.Api/Controllers/AccountsController.cs

```diff
@@ -206,12 +206,45 @@ public async Task<ActionResult<List<AccountDto>>> GetAllNotAssignedAccounts()
         /// Returns an updated account
         /// </summary>
         /// <response code="200">Successful return an updated account entity</response>
-        [Route("ChangePassword")]
+        [Route("password")]
         [Authorize(Roles = "Admin, Secretary, Mentor, Student")]
+        [HttpPut]
+        public async Task<ActionResult> ChangePassword(ChangeCurrentPasswordDto changePassword)
+        {
+            var updatedAccount = await _accountService.ChangePasswordAsync(changePassword);
+
+            return updatedAccount.ToActionResult();
+        }
+
+        /// <summary>
+        /// Returns a result of sending email
+        /// </summary>
+        /// <response code="200">Successful return a notification string</response>
+        [Route("password/forgot")]
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<ActionResult> ForgotPassword(ForgotPasswordDto changeForgotPassword)
+        {
+            if (await _accountService.IsEmailTakenAsync(changeForgotPassword.Email))
+            {
+                await _accountService.SendChangePasswordUrlAsync(changeForgotPassword);
+                
+                return Ok($"Link to change password has been sent to email {changeForgotPassword.Email}");
+            }
+
+            return BadRequest($"Account with email {changeForgotPassword.Email} does not exist!");
+        }
+
+        /// <summary>
+        /// Returns a result of confirmed password change
+        /// </summary>
+        /// <response code="200">Successful return an updated account entity</response>
+        [Route("password/reset/{guid}")]
+        [AllowAnonymous]
         [HttpPost]
-        public async Task<ActionResult> ChangePassword(ChangeCurrentPasswordDto changePasswd)
+        public async Task<ActionResult> ResetForgotPassword(string guid, ResetPasswordDto resetPassword)
```

Comment:
why guid type is string, not Guid?

Reply:
> why guid type is string, not Guid?

It comes from my decision when I've detected what we can't write guid to DB as guid. It's possible to write only as string.
So I decided to use string type for guid everywhere.
Don't worry, it doesn't make security issues.